### PR TITLE
To xml

### PR DESCRIPTION
**Pull request by EJV20**
Added the ToXML reactor with tests. Found out the RdbmsTableUploadReactor can't accept boolean values for imported data. Fixed that, however I am concerned with corner cases since I casted to RDBMSNativeEngine. I did look to see where import data method is used and noticed that it is used with only RDBMSNativeEngine, however I cannot change the method header from IDatabaseEngine due to its inheritance. 

Also added to the .gitignore to prevent running api tests from littering the change log.

### CODE REVIEW

File: src/prerna/reactor/database/upload/rdbms/RdbmsUploadReactorUtility.java [hunk 1]
```diff
@@ -106,7 +110,9 @@ public static String[] createNewTable(IDatabaseEngine engine, String tableName,
 			} else if (sType == SemossDataType.DATE) {
 				sqlTypes[i + 1] = "DATE";
 			} else if (sType == SemossDataType.TIMESTAMP) {
-				sqlTypes[i + 1] = "TIMESTAMP ";
```
Comment (EJV20) [inline src/prerna/reactor/database/upload/rdbms/RdbmsUploadReactorUtility.java:109]: Weird space after TIMESTAMP

File: src/prerna/reactor/database/upload/rdbms/RdbmsUploadReactorUtility.java [hunk 2]
```diff
@@ -91,6 +91,10 @@ public static String[] createNewTable(IDatabaseEngine engine, String tableName,
 		int size = types.length;
 		String[] sqlTypes = new String[size + 1];
 		String[] newHeaders = new String[size + 1];
+		
+		RDBMSNativeEngine rdbms = (RDBMSNativeEngine) engine;
+		AbstractSqlQueryUtil util = rdbms.getQueryUtil();
+		String bdtn = util.getBooleanDataTypeName();
```
Comment (EJV20) [inline src/prerna/reactor/database/upload/rdbms/RdbmsUploadReactorUtility.java]: Unsure why boolean was not supported

File: test/prerna/testing/ApiSemossTestEngineUtils.java [hunk 3]
```diff
@@ -45,7 +45,6 @@ public class ApiSemossTestEngineUtils {
 	private static Path ENGINES_CONFIG_FILE = Paths.get(ApiTestsSemossConstants.TEST_CONFIG_DIRECTORY.toString(), "engines.txt");
 	private static List<String> CORE_DBS = null;
 	
-	private static List<String> CURRENT_NAMES = new ArrayList<>();
```
Comment (EJV20) [inline test/prerna/testing/ApiSemossTestEngineUtils.java:48]: no longer checking current names, going to clean non core dbs after each test, unless if test class checks variable to not run the db cleaning method.

File: test/prerna/testing/ApiSemossTestUtils.java [hunk 4]
```diff
@@ -91,8 +91,29 @@ public static NounMetadata processRawPixel(String pixel) {
 		NounMetadata ret = pr.getResults().get(0);
 		return ret;
 	}
-
-	public static String buildPixelCall(Class<?> cl, Object... args) {
+
```
Comment (EJV20) [inline test/prerna/testing/ApiSemossTestUtils.java:94]: Allows for Pixel Chains with Select statements Database() | Select().as([]) | Iterate() | ToXml();

File: test/prerna/testing/reactor/export/ToXmlReactorTests.java [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+import prerna.reactor.export.ToXmlReactor;
+import prerna.reactor.qs.source.DatabaseReactor;
+import prerna.sablecc2.om.ReactorKeysEnum;
+import prerna.sablecc2.om.nounmeta.NounMetadata;
+import prerna.testing.AbstractBaseSemossApiTests;
+import prerna.testing.ApiSemossTestEngineUtils;
+import prerna.testing.ApiSemossTestInsightUtils;
+import prerna.testing.ApiSemossTestUtils;
+import prerna.testing.PixelChain;
+
```
Comment (EJV20) [inline test/prerna/testing/reactor/export/ToXmlReactorTests.java:28]: Testing Multiple rows, One row, Every data type supported (except factor). I believe Factor is somethign with R. Going to work on getting R tests running this week. Also added boolean support for table upload so I can export a boolean value here so I didn't have to write a new way to generate dbs.

File: .gitignore [hunk 6]
```diff
@@ -60,4 +60,15 @@ vector/**
...
+# test folder stuff
+testfolder/project/**
+testfolder/user/**
+testfolder/db/**
+testfolder/model/**
+testfolder/storage/**
+testfolder/vector/**
+testfolder/InsightCache/
+testfolder/mailpit/mailpit.log
+
```
Comment (EJV20) [inline .gitignore:74]: Preventing tests from littering changelog. Trying to increase ease of use and usability of testing environment over the coming weeks

Comment (EJV20) [reply .gitignore:74]: I think there is an inflection point where we have enough tests and utils, it becomes super simple to write any test and you only have to add one more piece of functionality to the test suite to write a super specific test.
